### ISSUE TITLE
DEFAULT values of DATE or TIMESTAMP can cause emulator to crash

### ISSUE THREAD

**Issue by nownabe** [GoogleCloudPlatform/cloud-spanner-emulator#115]
Hi. If a table has a default value that needs to be converted to its type, inserting a record will cause emulator to terminate.

This is an example of TIMESTAMP type.

```shell
$ SPANNER_EMULATOR_HOST=localhost:9010 spanner-cli -p test-project -i test-instance -d test-database
Connected.
spanner> CREATE TABLE some_table (id INT64 NOT NULL, timestamp TIMESTAMP NOT NULL DEFAULT ('1970-01-01T00:00:00Z')) PRIMARY KEY (id);
Query OK, 0 rows affected (0.00 sec)

spanner> INSERT INTO some_table (id) VALUES (1);
ERROR: spanner: code = "DeadlineExceeded", desc = "context deadline exceeded"
spanner>  
```

```shell
$ docker run -ti --rm -p 9010:9010 gcr.io/cloud-spanner-emulator/emulator:1.5.4
WARNING: proto: file "google/rpc/status.proto" is already registered
        previously from: "google.golang.org/genproto/googleapis/rpc/status"
        currently from:  "unknown"
See https://protobuf.dev/reference/go/faq#namespace-conflict

WARNING: proto: file "google/rpc/status.proto" has a name conflict over google.rpc.Status
        previously from: "google.golang.org/genproto/googleapis/rpc/status"
        currently from:  "unknown"
See https://protobuf.dev/reference/go/faq#namespace-conflict

WARNING: proto: message google.rpc.Status is already registered
        previously from: "google.golang.org/genproto/googleapis/rpc/status"
        currently from:  "unknown"
See https://protobuf.dev/reference/go/faq#namespace-conflict

2023/05/18 01:05:01 gateway.go:140: Cloud Spanner emulator running.
2023/05/18 01:05:01 gateway.go:141: REST server listening at 0.0.0.0:9020
2023/05/18 01:05:01 gateway.go:142: gRPC server listening at 0.0.0.0:9010
2023-05-18 01:05:38  FATAL  value.h : 605 : Check failed: ::absl::OkStatus() == ::zetasql::status_internal::AsStatus((value)) (OK vs. INTERNAL: ZETASQL_RET_CHECK failure (external/com_google_zetasql/zetasql/public/value.cc:249) field_type->Equals(value_type) 
Field type: TIMESTAMP
vs
Value type: STRING) 
2023/05/18 01:05:38 gateway.go:107: Shutting down gateway server since grpc server is terminated.
```

With Cloud Spanner, this works.

```shell
$ spanner-cli -p spanner-test-387100 -i test-instance -d test-database
Connected.
spanner> CREATE TABLE some_table (id INT64 NOT NULL, timestamp TIMESTAMP NOT NULL DEFAULT ('1970-01-01T00:00:00Z')) PRIMARY KEY (id);
Query OK, 0 rows affected (12.47 sec)

spanner> INSERT INTO some_table (id) VALUES (1);
Query OK, 1 rows affected (2.18 sec)
```

As long as a default value doesn't need to be converted, for example, `CURRENT_TIMESTAMP()` for TIMESTAMP type, emulator works properly.

DATE type also cause this problem. 

```
spanner> CREATE TABLE date_table (id INT64 NOT NULL, col_date DATE NOT NULL DEFAULT ("1970-01-01")) PRIMARY KEY (id);
Query OK, 0 rows affected (0.01 sec)

spanner> INSERT INTO date_table (id) VALUES (1);
ERROR: spanner: code = "DeadlineExceeded", desc = "context deadline exceeded"

// emulator output
2023-05-18 01:29:08  FATAL  value.h : 605 : Check failed: ::absl::OkStatus() == ::zetasql::status_internal::AsStatus((value)) (OK vs. INTERNAL: ZETASQL_RET_CHECK failure (external/com_google_zetasql/zetasql/public/value.cc:249) field_type->Equals(value_type) 
Field type: DATE
vs
Value type: STRING) 
```

If DATE and TIMESTAMP are in ARRAY, the problem doesn't happen.

```
spanner> CREATE TABLE array_table (id INT64 NOT NULL, col_array ARRAY<DATE> NOT NULL DEFAULT (ARRAY<DATE>['1970-01-01'])) PRIMARY KEY (id);
Query OK, 0 rows affected (0.01 sec)

spanner> INSERT INTO array_table (id) VALUES (1);
Query OK, 1 rows affected (0.00 sec)

spanner> SELECT * FROM array_table;
+----+--------------+
| id | col_array    |
+----+--------------+
| 1  | [1970-01-01] |
+----+--------------+
1 rows in set (570.031us)

spanner> CREATE TABLE array_table2 (id INT64 NOT NULL, col_array ARRAY<TIMESTAMP> NOT NULL DEFAULT (ARRAY<TIMESTAMP>['1970-01-01T00:00:00Z'])) PRIMARY KEY (id);
Query OK, 0 rows affected (0.00 sec)

spanner> INSERT INTO array_table2 (id) VALUES (1);
Query OK, 1 rows affected (0.00 sec)

spanner> SELECT * FROM array_table2;
+----+------------------------+
| id | col_array              |
+----+------------------------+
| 1  | [1970-01-01T00:00:00Z] |
+----+------------------------+
1 rows in set (614.939us)
```

**Comment by nownabe** [comment]
NUMERIC also cuases this problem.

```
spanner> CREATE TABLE some_table (id INT64 NOT NULL, col_numeric NUMERIC DEFAULT (1.23)) PRIMARY KEY (id);
Query OK, 0 rows affected (0.00 sec)

spanner> INSERT INTO some_table (id) VALUES (1);
ERROR: spanner: code = "Canceled", desc = "context canceled"

// emulator output
2023-05-18 02:31:20  FATAL  value.h : 576 : Check failed: ::absl::OkStatus() == ::zetasql::status_internal::AsStatus((value)) (OK vs. INTERNAL: ZETASQL_RET_CHECK failure (external/com_google_zetasql/zetasql/public/value.cc:252) field_type->Equals(value_type) 
Field type: NUMERIC
vs
Value type: DOUBLE) 
```

If a default value is defined with `NUMERIC` keyword like `DEFAULT (NUMERIC '1.23')`, it doesn't cause the problem.

**Comment by gauravpurohit06** [comment]
Thank you @nownabe for filing the issue. The issue has been fixed in the[ v@1.5.5](https://github.com/GoogleCloudPlatform/cloud-spanner-emulator/releases/tag/v1.5.5) of emulator.
